tests/UnleashBuilderTest.php

```diff
@@ -339,28 +339,36 @@ public function testWithoutDefaultClients()
         $cacheProperty = (new ReflectionObject($configuration))->getProperty('cache');
         $cacheProperty->setAccessible(true);
 
-        $unleash = $instance->build();
-        $repository = $repositoryProperty->getValue($unleash);
-        $configuration = $configurationProperty->getValue($repository);
-        $cache = $cacheProperty->getValue($configuration);
-
-        self::assertInstanceOf(FilesystemCachePool::class, $cache);
-
         $defaultImplementationsProperty = (new ReflectionObject($locator))->getProperty('defaultImplementations');
         $defaultImplementationsProperty->setAccessible(true);
         $defaultImplementations = $defaultImplementationsProperty->getValue($locator);
 
-        $defaultImplementations['cache'][FilesystemCachePool::class . 2] = [];
-        unset($defaultImplementations['cache'][FilesystemCachePool::class]);
-        $defaultImplementationsProperty->setValue($locator, $defaultImplementations);
+        // cache/filesystem-adapter should be used by default if it's installed
+        if (class_exists(FilesystemCachePool::class)) {
```

Comment:
This looks like a lot of changes, but it's just rearranging existing code and only running `cache/filesystem-adapter`-related tests if it's installed.